internal/management/controller/slots/infrastructure/replicationslot.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package infrastructure
+
+// SlotType represents the type of replication slot
+type SlotType string
+
+// SlotTypePhysical represents the physical replication slot
+const SlotTypePhysical SlotType = "physical"
+
+// ReplicationSlot represents a single replication slot
+type ReplicationSlot struct {
+	SlotName   string   `json:"slotName,omitempty"`
+	Type       SlotType `json:"type,omitempty"`
+	Active     bool     `json:"active"`
+	RestartLSN string   `json:"restartLSN,omitempty"`
+}
+
+// ReplicationSlotList contains a list of replication slots
+type ReplicationSlotList struct {
```

Comment:
The more I look at this, the more I think it should be a map[string]ReplicationSlot.
I.e. we should not allow, nor ever have, two slots with the same name.